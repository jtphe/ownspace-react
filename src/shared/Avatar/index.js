import React from 'react';
import { Image } from 'react-native';
import { CLIENT_COLOR_PRIMARY } from '@constants';

/**
 * The Avatar component
 * @param {string} image - URL or URI of the image
 * @param {number} size - The avatar's size
 * @param {number} borderRadius - The avatar's border radius
 */
const Avatar = ({ image, size, borderRadius }) => {
  /**
   * Get the avatar style
   */
  const _typeStyle = () => {
    return {
      width: size,
      height: size,
      borderRadius,
      borderColor: CLIENT_COLOR_PRIMARY,
      borderWidth: 1
    };
  };

  /**
   * Render the Avatar component
   * @returns {React.Component} - The Avatar component
   */
  return (
    <Image
      source={{
        uri:
          image === null ? 'http://eliseretouches.com/media/avatar.png' : image
      }}
      style={[_typeStyle()]}
    />
  );
};

export default Avatar;
