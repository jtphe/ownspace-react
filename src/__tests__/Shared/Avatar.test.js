import { Image } from 'react-native';
import React from 'react';
import Avatar from '@shared/Avatar';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import * as Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

describe('Avatar component tests', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Avatar />);
  });

  it('Avatar snapshot', () => {
    const tree = renderer.create(<Avatar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Image props', () => {
    const avatarImage = wrapper.find(Image).get(0);
    expect(avatarImage.props.source.uri).toEqual(
      'http://eliseretouches.com/media/avatar.png'
    );
    expect(avatarImage.props.style).toEqual({
      width: 100,
      height: 100,
      borderRadius: 50,
      borderColor: '#003466',
      borderWidth: 1
    });
  });
});

Enzyme.configure({ adapter: new Adapter() });
