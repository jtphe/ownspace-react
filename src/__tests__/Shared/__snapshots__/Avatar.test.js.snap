// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component tests Avatar snapshot 1`] = `
<Image
  source={
    Object {
      "uri": "http://eliseretouches.com/media/avatar.png",
    }
  }
  style={
    Object {
      "borderColor": "#003466",
      "borderRadius": 50,
      "borderWidth": 1,
      "height": 100,
      "width": 100,
    }
  }
/>
`;
